lib/track.lua

```diff
@@ -0,0 +1,73 @@
+local Track = {}
+
+function Track:new() 
+  local t = setmetatable({}, { 
+    __index = Track,
+    __tostring = function(t) return t:to_string() end
+  })
+  t.pattern = {}
+  t.pattern_string = ""
+  t.length = 8
+  t.division = 1
+  t.selected = false   -- for ui, not sequencing
+  t.selected_index = 1 -- ditto
+  return t
+end
+
+function Track:get_selected_index()
+  return self.selected_index
+end
+
+function Track:set_selected_index(n)
+  self.selected_index = util.clamp(n, 1, self:get_length())
+end
+
+function Track:to_string()
+  return self.pattern_string
+end
+
+function Track:set_selected_index(n)
+  self.selected_index = util.clamp(n, 1, #self.pattern)
+end
+
+function Track:change()
+  local new_pattern = self.pattern
+  new_pattern[self:get_selected_index()] = not new_pattern[self:get_selected_index()]
+  self:set_pattern(new_pattern)
+end
+
+function Track:set_divsion(n)
+  self.division = n
+end
+
+function Track:set_pattern(pattern)
+  self.length = #pattern
+  self.pattern = {}
+  self.pattern_string = ""
+  for i = 1, #pattern do
+    self.pattern[i] = pattern[i]
+    self.pattern_string = self.pattern_string .. (pattern[i] and "x" or "-")
```

Comment:
Will we want a way to do e.g. dynamics, pitch, parameter changes, bursts later on? Or maybe these can be in separate sequencers that would get attached to these?

Reply:
Would be nice to have per step parameter locking.

Reply:
I think so too - maybe we could come up with a nice, simple step container to hold these properties.